dace/transformation/interstate/continue_to_condition.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2019-2024 ETH Zurich and the DaCe authors. All rights reserved.
+import copy
+
+from dace import sdfg as sd, properties
+from dace.sdfg import utils as sdutil
+from dace.sdfg.state import ContinueBlock, ControlFlowRegion, ConditionalBlock
+from dace.transformation import transformation as xf
+
+
+@properties.make_properties
+@xf.explicit_cf_compatible
+class ContinueToCondition(xf.MultiStateTransformation):
+    """
+    Converts a continue statement in a loop to a condition in the loop.
+    """
+
+    cb = xf.PatternNode(ContinueBlock)
+
+    @classmethod
+    def expressions(cls):
+        return [sdutil.node_path_graph(cls.cb)]
+
+    def can_be_applied(self, graph, expr_index, sdfg, permissive=False):
+        assert isinstance(self.cb, ContinueBlock)
+
+        pg = self.cb.parent_graph.parent_graph
+
+        if not isinstance(pg, ConditionalBlock):
+            return False
+        if len(pg.branches) != 1:
+            return False
+
+        code, cfg = pg.branches[0]
+        if len(cfg.nodes()) != 1:
+            return False
+        if cfg.nodes()[0] is not self.cb:
+            return False
+
+        outer_cfg = pg.parent_graph
+        if len(outer_cfg.successors(pg)) > 1:
+            return False
+
+        return True
+
+    def apply(self, graph: ControlFlowRegion, sdfg: sd.SDFG):
+        # If there are no successors, we can just remove the continue node
```

Comment:
No, certainly not always! A continue block and the conditional it is contained in and pattern matched to can be nested N levels deep inside of a loop and / or further conditionals. That means that you need to check for any 'logical' successors too. This is only applicable if the conditional is nested _directly_ inside of a loop region, and the conditional has no successors.